[GTK] Use std::array and ASCIILiteral in KeyBindingTranslator.cpp

aperezdc opened the pull request:
#### 57c6ca3ab9ef6ce3b417d127ff0f592ce6493adc
<pre>
[GTK] Use std::array and ASCIILiteral in KeyBindingTranslator.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=282170">https://bugs.webkit.org/show_bug.cgi?id=282170</a>

Reviewed by Michael Catanzaro.

Convert gtkDeleteCommands, gtkMoveCommands, customKeyBindings, and
predefinedKeyBindings to use std::array, containing ASCIILiteral values
where appropriate.

* Source/WebKit/UIProcess/gtk/KeyBindingTranslator.cpp:
(WebKit::deleteFromCursorCallback): Remove comment about array bounds
checking now that they are using std::array.
(WebKit::handleKeyBindingsForMap): Receive a std::span instead of a raw
pointer and item count.
(WebKit::handleCustomKeyBindings): Adapt to pass a std::span.
(WebKit::KeyBindingTranslator::commandsForKeyval): Ditto.

Canonical link: <a href="https://commits.webkit.org/285804@main">https://commits.webkit.org/285804@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b252a904961063b240d36d441ee98f6740654472

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73868 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53297 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26679 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78199 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25106 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62430 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1082 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/58078 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16436 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76935 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48235 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63557 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38474 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45045 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23439 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66601 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21394 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79757 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1185 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/66410 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1328 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63570 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65690 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16249 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9583 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/7765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1149 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/3899 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/1178 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1165 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1184 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/gtk/KeyBindingTranslator.cpp:
```diff
@@ -82,25 +82,22 @@ static void showHelpCallback(GtkWidget* widget, KeyBindingTranslator*)
+    { "DeleteToBeginningOfLine"_s,      "DeleteToEndOfLine"_s      }, // Line ends
+    { "DeleteToBeginningOfParagraph"_s, "DeleteToEndOfParagraph"_s }, // Paragraph ends
+    { "DeleteToBeginningOfParagraph"_s, "DeleteToEndOfParagraph"_s }, // Paragraphs
+    { 0,                                0                          } // Whitespace (M-\ in Emacs)
```
Does nullptr not work here?

```suggestion
    { nullptr,                          nullptr                    } // Whitespace (M-\ in Emacs)
```

I think it should, because you must be using the `std::nullptr_t` constructor. Same for `gtkMoveCommands` below.

aperezdc replied:
Yes, this works as expected. Will change those before landing.

mcatanzaro reviewed (approved):
OK, I was confused at first, but I see this works because there is an implicit conversion from ASCIILiteral to `const char *`. When you pass 0, it will convert to an empty string. Pretty nice tbh.

aperezdc committed:
[GTK] Use std::array and ASCIILiteral in KeyBindingTranslator.cpp
https://bugs.webkit.org/show_bug.cgi?id=282170

Reviewed by Michael Catanzaro.

Convert gtkDeleteCommands, gtkMoveCommands, customKeyBindings, and
predefinedKeyBindings to use std::array, containing ASCIILiteral values
where appropriate.

* Source/WebKit/UIProcess/gtk/KeyBindingTranslator.cpp:
(WebKit::deleteFromCursorCallback): Remove comment about array bounds
checking now that they are using std::array.
(WebKit::handleKeyBindingsForMap): Receive a std::span instead of a raw
pointer and item count.
(WebKit::handleCustomKeyBindings): Adapt to pass a std::span.
(WebKit::KeyBindingTranslator::commandsForKeyval): Ditto.

Canonical link: https://commits.webkit.org/285804@main
